doc/release/1.17.0-notes.rst

```diff
@@ -43,6 +43,21 @@ from python.
 This deprecation should not affect many users since arrays created in such
 a manner are very rare in practice and only available through the NumPy C-API.
 
+Writing to the result of `numpy.broadcast_array` will warn
+----------------------------------------------------------
+
+Commonly `numpy.broadcast_array` returns a writeable array with internal
+overlap, making it unsafe to write to. A future version will set the
+``writeable`` flag to ``False``, and require users to manually set it to
+``True`` if they are sure that is what they want to do. From this version,
+writing to it will emit a deprecation warning with instructions to set the
+``writeable`` flag ``True``.  Note that if one were to inspect the flag before
+setting it, one would, perhaps confusingly, find it would already be ``True``.
+Explicitly setting it, though, as one will need to do in future versions,
+clears an internal flag that is used to produce the deprecation warning. An
+additional `FutureWarning` will be emitted when accessing the ``writeable``
```

Comment:
Upon looking at the PR again, I now see where this second future warning is emitted. Maybe delete the "perhaps confusingly" insert I suggested earlier and start this sentence with: "To help alleviate confusion, an additional ..."